SetupDataPkg/Library/ActiveProfileSelectorLibNull/ActiveProfileSelectorLibNull.c

```diff
@@ -0,0 +1,80 @@
+/** @file
+  Library interface to query the active configuration profile.
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+#include <Uefi.h>
+#include <Library/BaseLib.h>
+#include <Library/DebugLib.h>
+#include <Library/PcdLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/ActiveProfileSelectorLib.h>
+
+/**
+  Return which profile is the active profile for this boot.
+  This function validates the profile GUID is valid.
+
+  @param[out] ActiveProfileGuid   The file GUID for the active profile. Caller frees memory.
+                                  NULL is returned in case of failure.
+
+  @retval EFI_INVALID_PARAMETER   Input argument is null.
+  @retval EFI_NO_RESPONSE         The source of truth for profile selection has returned a garbage value or not replied.
+  @retval EFI_OUT_OF_RESOURCES    Memory allocation failed.
+  @retval EFI_SUCCESS             The operation succeeds.
+
+**/
+EFI_STATUS
+EFIAPI
+RetrieveActiveProfileGuid (

```

Comment:
Actually, now that I looked at the usage of this function, it might be better in the driver itself? This routine can then be called validation, instead of retrieving? Then this library can be a real null function and let the platform populate how this is done accordingly. (without duplicating this routine inside)

Reply:
Agreed, refactored into driver.